gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/matching/common/query/predicates/wrappers/ComparableWrapper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of Gradoop.
+ *
+ * Gradoop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Gradoop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Gradoop. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.gradoop.flink.model.impl.operators.matching.common.query.predicates.wrappers;
+
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.impl.operators.matching.common.query.predicates.wrappers
+  .comparables.ElementSelectorWrapper;
+import org.gradoop.flink.model.impl.operators.matching.common.query.predicates.wrappers
+  .comparables.LiteralWrapper;
+import org.gradoop.flink.model.impl.operators.matching.common.query.predicates.wrappers
+  .comparables.PropertySelectorWrapper;
+import org.gradoop.flink.model.impl.operators.matching.single.cypher.embeddings.EmbeddingEntry;
+import org.s1ck.gdl.model.comparables.ComparableExpression;
+import org.s1ck.gdl.model.comparables.ElementSelector;
+import org.s1ck.gdl.model.comparables.Literal;
+import org.s1ck.gdl.model.comparables.PropertySelector;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * Wrapps a {@link ComparableExpression}
+ */
+public abstract class ComparableWrapper implements Serializable {
```

Comment:
I am not a fan of naming classes after the pattern they implement. Can't we name it `ComparableExpression` or
use a prefix like `CypherExpression`?